device/tt_arch_types.h

```diff
@@ -10,6 +10,9 @@
 #include "device/architecture.h"
 
 namespace tt {
+
+// TODO: why do we have ARCH and architecture?  This is a mess.  Can we have just one?
+// Can we get rid of the entries that (for all practical purposes) do not exist?
```

Comment:
Yes, just so we don't forget, #161

Reply:
Thanks for adding that - I've been in the habit of leaving these TODOs as I am working on the code with the intent of creating actual issues for them, but some I forget about!